e2etests/tests/routes.go

```diff
@@ -0,0 +1,276 @@
+			Leaf: leaf,
+			Red: infra.Addresses{
+				IPV4: redPrefixes,
+			},
+			Blue: infra.Addresses{
+				IPV4: bluePrefixes,
+			},
+		}
+		config, err := infra.LeafConfigToFRR(leafConfiguration)
+		Expect(err).NotTo(HaveOccurred())
+		err = leaf.ReloadConfig(config)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	removeLeafPrefixes := func(leaf infra.Leaf) {
+		changeLeafPrefixes(leaf, []string{}, []string{})
+	}
+
+	BeforeAll(func() {
+		err := Updater.CleanAll()
+		Expect(err).NotTo(HaveOccurred())
+
+		cs = k8sclient.New()
+		routerPods, err = openperouter.RouterPods(cs)
+		Expect(err).NotTo(HaveOccurred())
+		frrk8sPods, err = frrk8s.Pods(cs)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = Updater.Update(config.Resources{
+			Underlays: []v1alpha1.Underlay{
+				infra.Underlay,
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterAll(func() {
+		err := Updater.CleanAll()
+		Expect(err).NotTo(HaveOccurred())
+		By("waiting for the router pod to rollout after removing the underlay")
+		Eventually(func() error {
+			return openperouter.DaemonsetRolled(cs, routerPods)
+		}, 2*time.Minute, time.Second).ShouldNot(HaveOccurred())
+
+		removeLeafPrefixes(infra.LeafAConfig)
+		removeLeafPrefixes(infra.LeafBConfig)
+	})
+
+	BeforeEach(func() {
+		err := Updater.CleanButUnderlay()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		err := Updater.CleanButUnderlay()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	Context("with vnis", func() {
+		BeforeEach(func() {
+			err := Updater.Update(config.Resources{
+				VNIs: []v1alpha1.VNI{
+					vniRed,
+					vniBlue,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			removeLeafPrefixes(infra.LeafAConfig)
+			removeLeafPrefixes(infra.LeafBConfig)
+		})
+
+		It("receives type 5 routes from the fabric", func() {
+			Contains := true
+			checkRouteForLeaf := func(leaf infra.Leaf, vni v1alpha1.VNI, mustContain bool, prefixes []string) {
+				Eventually(func() error {
+					for _, p := range routerPods {
+						exec := executor.ForPod(p.Namespace, p.Name, "frr")
+						evpn, err := frr.EVPNInfo(exec)
+						Expect(err).NotTo(HaveOccurred())
+						for _, prefix := range prefixes {
+							if mustContain && !evpn.ContainsType5RouteForVNI(prefix, leaf.VTEPIP, int(vni.Spec.VNI)) {
+								return fmt.Errorf("type5 route for %s - %s not found in %v in router %s", prefix, leaf.VTEPIP, evpn, p.Name)
+							}
+							if !mustContain && evpn.ContainsType5RouteForVNI(prefix, leaf.VTEPIP, int(vni.Spec.VNI)) {
+								return fmt.Errorf("type5 route for %s - %s found in %v in router %s", prefix, leaf.VTEPIP, evpn, p.Name)
+							}
+						}
+					}
+					return nil
+				}, 2*time.Minute, time.Second).ShouldNot(HaveOccurred())
+			}
+
+			By("announcing type 5 routes on VNI 100 from leafA")
+			changeLeafPrefixes(infra.LeafAConfig, leafAVRFRedPrefixes, emptyPrefixes)
+			checkRouteForLeaf(infra.LeafAConfig, vniRed, Contains, leafAVRFRedPrefixes)
+
```

Comment:
can we have a !Contains for blue here?

Reply:
we are not advertising those routes yet, so what it would be testing? Do you have a particular reason for wanting it?